blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/metrics/AbstractSerDes.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io.serializers.metrics;
+
+import com.google.protobuf.CodedInputStream;
+import com.google.protobuf.CodedOutputStream;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.types.*;
+
+import java.io.IOException;
+
+/**
+ * This is an abstract class that most Serialization/Deserialization class
+ * extends from. It contains some utility methods.
+ */
+public abstract class AbstractSerDes {
```

Comment:
Yeah, you are right it's not so much of an Abstract class. It's more of a Base class. The SerDes subclasses share common logic such as the size of size and type in the serialized buffer. I can change the name of the class from AbstractSerDes to BaseSerDes.

I had in mind of creating an interface. But as you noticed on classes like RawSerDes, it's hard to do it. Also, not all objects to be serialized extends or implements a common class/interface. Those objects such as Average, Min, Max do share a parent class, but the Rollup objects share a different parent class. Changing those objects to have a common inheritance is a massive change, I feel the cost not quite justified yet.